instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -629,34 +721,67 @@ async def __call__(
                     span_name,
                     scope,
                     send,
-                    duration_attrs,
+                    attributes,
                 )
 
                 await self.app(scope, otel_receive, otel_send)
         finally:
             if scope["type"] == "http":
                 target = _collect_target_attribute(scope)
                 if target:
-                    duration_attrs[SpanAttributes.HTTP_TARGET] = target
+                    path, query = _parse_url_query(target)
+                    _set_http_target(
+                        attributes,
+                        target,
+                        path,
+                        query,
+                        self._sem_conv_opt_in_mode,
+                    )
                 duration = max(round((default_timer() - start) * 1000), 0)
-                self.duration_histogram.record(duration, duration_attrs)
+                duration_attrs_old = _parse_duration_attrs(
+                    attributes, _HTTPStabilityMode.DEFAULT
+                )
+                if target:
+                    duration_attrs_old[SpanAttributes.HTTP_TARGET] = target
+                duration_attrs_new = _parse_duration_attrs(
+                    attributes, _HTTPStabilityMode.HTTP
+                )
+                if self.duration_histogram_old:
+                    self.duration_histogram_old.record(
+                        duration, duration_attrs_old
+                    )
+                if self.duration_histogram_new:
+                    self.duration_histogram_new.record(
+                        duration, duration_attrs_new
```

Comment:
The duration here is in seconds, right? But in L740 we are getting `duration` in ms. Does it make sense to use duration in seconds here?

Reply:
Good catch